lib/proximity.js

```diff
@@ -0,0 +1,249 @@
+var Board = require("../lib/board.js"),
+  events = require("events"),
+  util = require("util");
+
+function analogHandler(opts, dataHandler) {
+
+  this.io.pinMode(this.pin, this.io.MODES.ANALOG);
+
+  this.io.analogRead(this.pin, function(data) {
+    dataHandler.call(this, data);
+  }.bind(this));
+
+}
+
+var Controllers = {
+  GP2Y0A21YK: {
+    // https://www.sparkfun.com/products/242
+    initialize: {
+      value: analogHandler
+    },
+    toCm: {
+      value: function(raw) {
+        return +(12343.85 * Math.pow(raw, -1.15)).toFixed(2);
+      }
+    }
+  },
+  GP2D120XJ00F: {
+    // https://www.sparkfun.com/products/8959
+    initialize: {
+      value: analogHandler
+    },
+    toCm: {
+      value: function(raw) {
+        return +((2914 / (raw + 5)) - 1).toFixed(2);
+      }
+    }
+  },
+  GP2Y0A02YK0F: {
+    // https://www.sparkfun.com/products/8958
+    // 15cm - 150cm
+    initialize: {
+      value: analogHandler
+    },
+    toCm: {
+      value: function(raw) {
+        return +(10650.08 * Math.pow(raw, -0.935) - 10).toFixed(2);
+      }
+    }
+  },
+  GP2Y0A41SK0F: {
+    // https://www.sparkfun.com/products/12728
+    // 4cm - 30cm
+    initialize: {
+      value: analogHandler
+    },
+    toCm: {
+      value: function(raw) {
+        return +(2076 / (raw - 11)).toFixed(2);
+      }
+    }
+  },
+  SRF10: {
+    initialize: {
+      value: function(opts, dataHandler) {
+
+        var address = 0x70;
+        var delay = 65;
+
+        // Set up I2C data connection
+        this.io.i2cConfig(0);
+
+        // Startup parameter
+        this.io.i2cWrite(address, [0x01, 16]);
+        this.io.i2cWrite(address, [0x02, 255]);
+
+        // Is this still a thing?
+        //this.io.setMaxListeners(100);
+
+        function read() {
+          this.io.i2cWrite(address, [0x02]);
+          this.io.i2cRead(address, 2, function(data) {
```

Comment:
I tested this locally and the only change that needs to be made is here: 

`this.io.i2cRead` => `this.io.i2cReadOnce`

(This device can't be read continuously)